app/models/customer.rb

```diff
@@ -1,3 +1,15 @@
 class Customer < ApplicationRecord
   has_many :invoices
+  has_many :transactions, through: :invoices
+
+  def self.top_five_by_transaction_count
```

Comment:
I feel like you really did a number on this one. Almost two advanced AR stories out of the way.